src/game/game_text.hpp

```diff
@@ -18,21 +18,25 @@ void ReconsiderGameScriptLanguage();
 
 /** Container for the raw (unencoded) language strings of a language. */
 struct LanguageStrings {
-	const char *language; ///< Name of the language (base filename).
-	StringList lines;     ///< The lines of the file to pass into the parser/encoder.
+	std::string language; ///< Name of the language (base filename). Empty string if invalid.
+	StringList  lines;    ///< The lines of the file to pass into the parser/encoder.
 
-	LanguageStrings(const char *language, const char *end = nullptr);
-	~LanguageStrings();
+	LanguageStrings() {}
+	LanguageStrings(const std::string &lang) : language(lang) {}
+	LanguageStrings(const LanguageStrings &other) : language(other.language), lines(other.lines) {}
+	LanguageStrings(LanguageStrings &&other) : language(std::move(other.language)), lines(std::move(other.lines)) {}
+
+	bool IsValid() const { return !this->language.empty(); }
 };
 
 /** Container for all the game strings. */
 struct GameStrings {
-	uint version;                                  ///< The version of the language strings.
-	std::shared_ptr<LanguageStrings> cur_language; ///< The current (compiled) language.
+	uint version;                  ///< The version of the language strings.
+	LanguageStrings *cur_language; ///< The current (compiled) language.
```

Comment:
Where is this pointer owned?

Reply:
The `compiled_strings` vector manages the pointed-to object. We never remove loaded gs languages, as such there's no need for further tracking.

Reply:
Please declare copy/move operators/constructors as "deleted".